EIPS/eip-7928.md

```diff
@@ -0,0 +1,199 @@
+
+class AccountAccess(Container):
+    address: Address
+    accesses: List[SlotAccess, MAX_SLOTS]
+
+BlockAccessList = List[AccountAccess, MAX_ACCOUNTS]
+
+# Balance Diff structures
+BalanceDelta = ByteVector(11)  # signed, two's complement encoding
+
+class BalanceChange(Container):
+    tx_index: TxIndex
+    delta: BalanceDelta  # signed integer, encoded as 11-byte vector
+
+class AccountBalanceDiff(Container):
+    address: Address
+    changes: List[BalanceChange, MAX_TXS]
+
+BalanceDiffs = List[AccountBalanceDiff, MAX_ACCOUNTS]
+
+# Post-block nonce structures
+class AccountNonce(Container):
+    address: Address  # account address
+    nonce_after: Nonce  # nonce value after block execution
+
+NonceDiffs = List[AccountNonceDiff, MAX_TXS]
+```
+
+The `BlockAccessList` is a deduplicated list of accessed addresses. For each address, it MUST contain a list of accessed storage keys. 
+For writes, each `SlotAccess` MUST contain an ordered list of transaction indices that accessed this key, and the final storage value after the last access. Transactions with writes that do not change the storage value MUST NOT contain a `value_after`.
+For contract deployments, the `value_after` MUST contain the runtime bytecode of the newly deployed contract.
+
+For reads, each `SlotAccess` MUST contain an empty `accesses` list.
+
+The `BalanceDiffs` structure tracks every address with a balance change, including transaction senders, recipients, and the block's coinbase address. 
+Each entry MUST include the transaction index and the signed balance delta per address for each transaction.
+11 bytes are sufficient to represent the total ETH supply.
+
+The `NonceDiffs` structure MUST record the post-block nonce values for all `CREATE2` deployer accounts and the deploayed contracts in the block.
+
+### State Transition Function
+
+Modify the state transition function to validate the block-level access lists:
+
+```python 
+def state_transition(block: Block) -> None:
+    computed_access_list = {}
+    computed_balance_diffs = {}
+    computed_nonce_diffs = {}
+
+    for idx, tx in enumerate(block.transactions):
+        # Execute transaction and track state accesses
+        accessed_items, balance_changes = execute_transaction(tx)
+        
+        # Process storage accesses
+        for (address, slot), is_write, value in accessed_items:
+            if (address, slot) not in computed_access_list:
+                computed_access_list[(address, slot)] = []
+            
+            access_entry = {"tx_index": idx}
+            if is_write:
+                access_entry["value_after"] = value
+            
+            computed_access_list[(address, slot)].append(access_entry)
+        
+        # Process balance changes
+        for address, delta in balance_changes:
+            if address not in computed_balance_diffs:
+                computed_balance_diffs[address] = []
+            computed_balance_diffs[address].append({"tx_index": idx, "delta": delta})
```

Comment:
All looks fine but we can likely take out a lot of items for default behavior. (Of course we should still account for the worst case scenario).

Reply:
yeah the STF is still a rough draft. I will update it as soon as I get to spec'ing it.